coerces numeric strings into numbers

The current implementation does not strictly coerce numeric strings. Node assert(), is itself coercive and is matching the numeric string to the integer. 

```
  // This stricter test fails, as the object n.value is actually a string.

  it("coerces numeric strings into numbers (for free)", function(){
    var n = fm(fixtures.numeric)
    assert.strictEqual(n.integral, 10000000)
    assert.strictEqual(n.decimal, 3.1415)
    assert.strictEqual(n.negative, -100)
    assert.strictEqual(n.stringy, "I am not a number")
  })

1 failing

  1) html-frontmatter coerces numeric strings into numbers (for free):
     AssertionError: "10000000" === 10000000

```

--- a/index.js
+++ b/index.js
@@ -27,6 +27,9 @@ var parse = module.exports = function(input) {
       if (value === "true") value = true
       if (value === "false") value = false
 
+      var num = +value // resolves to number or NaN
+      if (num) value = num
+
       obj[key] = value
     })
 

--- a/test/index.js
+++ b/test/index.js
@@ -57,10 +57,10 @@ describe("html-frontmatter", function() {
 
   it("coerces numeric strings into numbers (for free)", function(){
     var n = fm(fixtures.numeric)
-    assert.equal(n.integral, 10000000)
-    assert.equal(n.decimal, 3.1415)
-    assert.equal(n.negative, -100)
-    assert.equal(n.stringy, "I am not a number")
+    assert.strictEqual(n.integral, 10000000)
+    assert.strictEqual(n.decimal, 3.1415)
+    assert.strictEqual(n.negative, -100)
+    assert.strictEqual(n.stringy, "I am not a number")
   })
 
   // Convenience

--- a/test/index.js
+++ b/test/index.js
@@ -55,7 +55,7 @@ describe("html-frontmatter", function() {
     assert.deepEqual(fm(fixtures.boolean), {good: true, bad: false})
   })
 
-  it("coerces numeric strings into numbers (for free)", function(){
+  it("coerces numeric strings into numbers", function(){
     var n = fm(fixtures.numeric)
     assert.strictEqual(n.integral, 10000000)
     assert.strictEqual(n.decimal, 3.1415)

